src/main/kotlin/org/flaxo/plagiarism/data2graph.kt

```diff
@@ -199,6 +205,12 @@ fun Viz.refreshGraph(nodes: List<GraphNode>, links: List<GraphLink>, simulation:
     simulation.addForce("Graph force", graphForce)
```

Comment:
Did you try to filter links by their nodes visibility while creating `GraphForce`? It could probably help to optimize graph if some of the nodes are omitted.
```kotlin
...
val activeLinks = links.filter { it.first in visibleNames && it.second in visibleNames }
val graphForce = GraphForce(activeLinks, nodes, normalization, scale, shift)
...
```
It may not work the way I expect but it deserves a try.

Reply:
It just breaks everything :)